### PR TITLE
Updating Github templates

### PR DESCRIPTION
**Pull request by wolfenrain**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this MR is changing it, and what motivated the change.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The Flame analyzer (`./scripts/analyze.sh`) does not report any problems on my PR.
- [ ] I have added examples for new features in `doc/examples`.
- [ ] I have formatted my code with `./scripts/format.sh`.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [ ] I am willing to follow-up on review comments in a timely manner.
- [ ] I am done with this PR and removed the `Draft` status, by clicking on the `Read for review` button in this PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [ ] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/1_bug.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+name: Bug report.
+about: You are creating a Game with Flame but you are noticing strange behaviour, throws an exception, or it is buggy.
```
Comment (luanpotter) [inline .github/ISSUE_TEMPLATE/1_bug.md:3]: nit: noticing some strange behaviour, that it throws an unexpected exception, or that it is not working according to the specifications.

File: .github/ISSUE_TEMPLATE/2_feature.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+# Problem to solve
+<!-- Which problem would be solved with this feature? -->
+
+# Proposal
+<!-- What do you propose as a solution? Add as much information as you can! -->
+
+# More information
+<!-- Do you have any other useful information about this feature report? Please write it down here -->
+<!-- Possible helpful information: references to other sites/repositories -->
```
Comment (luanpotter) [inline .github/ISSUE_TEMPLATE/2_feature.md:20]: are you interesting in working on a PR for this?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,25 +1,36 @@
-# Description
+## Description
 
-Please include a summary of the change.
+*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this MR is changing it, and what motivated the change.*
```
Comment (luanpotter) [inline .github/pull_request_template.md:3]: In particular, if this is a breaking change, specify this here and what APIs where changed.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,25 +1,36 @@
...
+- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
+- [ ] All existing and new tests are passing.
+- [ ] I updated/added relevant documentation (doc comments with `///`).
+- [ ] The Flame analyzer (`./scripts/analyze.sh`) does not report any problems on my PR.
+- [ ] I have added examples for new features in `doc/examples`.
+- [ ] I have formatted my code with `./scripts/format.sh`.
+- [ ] I read and followed the [Flutter Style Guide].
+- [ ] I have added a description of the change under `[next]` in `CHANGELOG.md`.
+- [ ] I am willing to follow-up on review comments in a timely manner.
+- [ ] I am done with this PR and removed the `Draft` status, by clicking on the `Read for review` button in this PR.
```
Comment (luanpotter) [inline .github/pull_request_template.md:19]: I am done sounds a bit too profound here, how about something like:
`I am happy with the current version of this PR to be reviewed` or something

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -1,25 +1,36 @@
...
 
-- [ ] Bug fix (non-breaking change which fixes an issue)
-- [ ] New feature (non-breaking change which adds functionality)
-- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-- [ ] This change requires a documentation update
+## Breaking Change
 
-## Checklist:
+Does your PR require plugin users to manually update their apps to accommodate your change?
```
Comment (luanpotter) [inline .github/pull_request_template.md:23]: Flame users? Had to think a little bit to associate flame with plugin

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+# Contributing Guidelines
+If you're interested in contributing to this project, here are a few ways to do so:
+
+## Bug fixes
+* If you find a bug, please first report it using [Github issues](https://github.com/flame-engine/flame/issues/new).
```
Comment (luanpotter) [inline CONTRIBUTING.md:5]: first check if there is not already an issue; duplicated issues will be closed.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+* Issues that have already been identified as a bug will be labelled "bug".
+* If you'd like to submit a fix for a bug, send a [Pull Request](https://guides.github.com/activities/forking/#making-a-pull-request) from your own fork, also read the [How To](#how-to) and [Development Guidelines](#development-guidelines).
+* Include a test that isolates the bug and verifies that it was fixed.
+* Also update the example and documentation if necessary.
+
+## New Features
+* If you'd like to add a feature to the library that doesn't already exist, feel free to describe the feature in a new [Github issues](https://github.com/flame-engine/flame/issues/new).
```
Comment (luanpotter) [inline CONTRIBUTING.md:12]: if it's a bigger change, you can also get some initial thoughts about the general idea on discord beforehand.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+# How To
+* First of all [file an bug or feature report](https://github.com/flame-engine/flame/issues/new) on this repository.
+* [Fork the project](https://guides.github.com/activities/forking/#fork) on Github
+* Clone the forked repository to your local development machine (e.g. `git clone https://github.com/<YOUR_GITHUB_USER>/flame.git`)
+* Run `pub get` in the cloned repository to get all the dependencies
+* Create a new local branch based on issue number from first step (e.g. `git checkout -b 12-new-feature`)
+* Make your changes
+* When committing your changes, make sure to start the commit message with `#<issue-number>` (e.g. `git commit -m '#12 - New Feature added'`)
+* Push your new branch to your own fork into the same remote branch (e.g. `git push origin 12-new-feature`)
+* On Gitlab goto the [pull request page](https://guides.github.com/activities/forking/#making-a-pull-request) on your own fork and create a merge request to this reposistory
```
Comment (luanpotter) [inline CONTRIBUTING.md:42]: is that subliminal message to us or something?

Comment (luanpotter) [reply CONTRIBUTING.md:42]: :p

File: scripts/analyze.sh [hunk 9]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
```
Comment (luanpotter) [inline scripts/analyze.sh:1]: this was like this before but I normally like to skip a line between the shebang and the rest of the lines

File: CONTRIBUTING.md (hunk 8 above)
Comment (wolfenrain) [reply CONTRIBUTING.md:42]: Yes, yes it was :D

**Review by luanpotter — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
+# Contributing Guidelines
+If you're interested in contributing to this project, here are a few ways to do so:
```
Comment (luanpotter) [inline CONTRIBUTING.md:2]: followup: if rules are not followed we will close the issue/pr. please follow the instructions carefully.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+## Bug fixes
+* If you find a bug, please first report it using [Github issues](https://github.com/flame-engine/flame/issues/new).
+  * First check if there is not already an issue for it; duplicated issues will be closed.
+* Issues that have already been identified as a bug will be labelled "bug".
+* If you'd like to submit a fix for a bug, send a [Pull Request](https://guides.github.com/activities/forking/#making-a-pull-request) from your own fork, also read the [How To](#how-to) and [Development Guidelines](#development-guidelines).
+* Include a test that isolates the bug and verifies that it was fixed.
+* Also update the example and documentation if necessary.
```
Comment (luanpotter) [inline CONTRIBUTING.md:10]: followup nit: examples

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+## New Features
+* If you'd like to add a feature to the library that doesn't already exist, feel free to describe the feature in a new [Github issues](https://github.com/flame-engine/flame/issues/new).
+  * You can also join us on [Discord](https://discord.gg/pxrBmy4) to discuss some initials thoughts.
+* Issues that have been identified as a feature request will be labelled "feature".
+* If you'd like to implement the new feature, please wait for feedback from the project maintainers before spending too much time writing the code. In some cases, enhancements may not align well with the project objectives at the time.
+* Implement your code and please read the [How To](#how-to) and [Development Guidelines](#development-guidelines).
+* Also update the examples and documentation where needed.
+
+## Documentation & Miscellaneous
+* If you think the documentation could be clearer, or you have an alternative implementation of something that may have more advantages, we would love to hear it.
```
Comment (luanpotter) [inline CONTRIBUTING.md:21]: or an example?

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+# Requirements
+For a contribution to be accepted:
+
+* Take note of the [Development Guidelines](#development-guidelines)
+* Code must follow existing styling conventions
+* Commit message should start with a [issue number](#how-to) and should also be descriptive.
+
+If the contribution doesn't meet these criteria, a maintainer will discuss it with you on the issue. You can still continue to add more commits to the branch you have sent the Pull Request from.
+
+# How To
```
Comment (luanpotter) [inline CONTRIBUTING.md:35]: How to make a PR?

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+* First of all [file an bug or feature report](https://github.com/flame-engine/flame/issues/new) on this repository.
+* [Fork the project](https://guides.github.com/activities/forking/#fork) on Github
+* Clone the forked repository to your local development machine (e.g. `git clone https://github.com/<YOUR_GITHUB_USER>/flame.git`)
+* Run `pub get` in the cloned repository to get all the dependencies
```
Comment (luanpotter) [inline CONTRIBUTING.md:39]: flutter pub get

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+* Create a new local branch based on issue number from first step (e.g. `git checkout -b 12-new-feature`)
+* Make your changes
+* When committing your changes, make sure to start the commit message with `#<issue-number>` (e.g. `git commit -m '#12 - New Feature added'`)
+* Push your new branch to your own fork into the same remote branch (e.g. `git push origin 12-new-feature`)
+* On Github goto the [pull request page](https://guides.github.com/activities/forking/#making-a-pull-request) on your own fork and create a merge request to this reposistory
+
+# Development Guidelines
+* Documentation should be updated.
+* Example application should be updated.
```
Comment (luanpotter) [inline CONTRIBUTING.md:48]: examples

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+* If you'd like to add a feature to the library that doesn't already exist, feel free to describe the feature in a new [Github issues](https://github.com/flame-engine/flame/issues/new).
+  * You can also join us on [Discord](https://discord.gg/pxrBmy4) to discuss some initials thoughts.
+* Issues that have been identified as a feature request will be labelled "enchanment".
+* If you'd like to implement the new feature, please wait for feedback from the project maintainers before spending too much time writing the code. In some cases, enhancements may not align well with the project objectives at the time.
+* Implement your code and please read the [How To](#how-to) and [Development Guidelines](#development-guidelines).
+* Also update the examples and documentation where needed.
+
+## Documentation & Miscellaneous
+* If you think the documentation could be clearer or an example should be added, or you have an alternative implementation of something that may have more advantages, we would love to hear it.
+* As always first file a report in a [Github issues](https://github.com/flame-engine/flame/issues/new).
```
Comment (spydon) [inline CONTRIBUTING.md:24]: Remove `a`

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+* Issues that have been identified as a documentation change will be labelled "documentation".
+* Implement the changes to the documentation, please read the [How To](#how-to) and [Development Guidelines](#development-guidelines).
+
+# Requirements
+For a contribution to be accepted:
+
+* Take note of the [Development Guidelines](#development-guidelines)
+* Code must follow existing styling conventions
+* Commit message should start with a [issue number](#how-to) and should also be descriptive.
```
Comment (spydon) [inline CONTRIBUTING.md:33]: I think we should go with conventional commits here, but we can change to that later.

**Review by spydon — APPROVED** [review]

File: .github/pull_request_template.md [hunk 18]
```diff
@@ -1,25 +1,37 @@
-# Description
+## Description
```
[the remark this answers is not in the document]
Comment (wolfenrain) [reply .github/pull_request_template.md:1]: It is not? It is just a header indication right? I havent seen any linter that complains about this.

File: CONTRIBUTING.md (hunk 17 above)
Comment (wolfenrain) [reply CONTRIBUTING.md:33]: Yes agreed, but we dont have that yet :(

File: .github/pull_request_template.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (spydon) [reply .github/pull_request_template.md:1]: My IDE linter complains at least :shrug:

File: .github/pull_request_template.md [hunk 19]
```diff
@@ -1,25 +1,37 @@
...
+- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
+- [ ] All existing and new tests are passing.
+- [ ] I updated/added relevant documentation (doc comments with `///`).
+- [ ] The Flame analyzer (`./scripts/analyze.sh`) does not report any problems on my PR.
+- [ ] I have added examples for new features in `doc/examples`.
+- [ ] I have formatted my code with `./scripts/format.sh`.
+- [ ] I read and followed the [Flutter Style Guide].
+- [ ] I have added a description of the change under `[next]` in `CHANGELOG.md`.
+- [ ] I am willing to follow-up on review comments in a timely manner.
+- [ ] I am happy with the current version of this PR and it is ready to be reviewd
```
Comment (spydon) [inline .github/pull_request_template.md:19]: reviewed

File: .github/pull_request_template.md [hunk 20]
```diff
@@ -1,25 +1,37 @@
...
+  - [ ] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
```
Comment (spydon) [inline .github/pull_request_template.md:20]: Should this be indented? Looked a bit weird in the live version

File: .github/pull_request_template.md [hunk 21]
```diff
@@ -1,25 +1,37 @@
...
+*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this MR is changing it, and what motivated the change. If this is a breaking change, specify explicitly which APIs have been changed*
 
-Fixes # (issue number, if there is one)
+## Checklist
 
-## Type of change
+Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.
 
-Please delete options that are not relevant.
+- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
```
Comment (spydon) [inline .github/pull_request_template.md:9]: Too many boxes! The cats would love it though

File: .github/pull_request_template.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (wolfenrain) [reply .github/pull_request_template.md:1]: maybe we should add a markdown linter in the CI/CD. Also for our docs and dead links :-)

File: .github/pull_request_template.md (hunk 21 above)
Comment (wolfenrain) [reply .github/pull_request_template.md:9]: Miaauw :cat:

File: .github/pull_request_template.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (spydon) [reply .github/pull_request_template.md:1]: On pubdev it now looks weird with my added empty #, so it is probably not correct that way either 😅

**Pull request merged**